### PR TITLE
wrap external components

### PR DESCRIPTION
**Pull request by zarsky-broad**

### CODE REVIEW

File: src/pages/workspaces/List.js [hunk 1]
```diff
@@ -160,14 +160,13 @@ const WorkspaceList = hh(class WorkspaceList extends Component {
...
-          _.isEmpty(workspaces) ?
-            'You don\'t seem to have access to any workspaces.' :
-            listView ?
-              this.wsList() : this.wsGrid()
+        Utils.cond(
+          [failure, `Couldn't load workspace list: ${failure}`],
+          [!workspaces, spinner({ size: 64 })],
+          [_.isEmpty(workspaces), 'You don\'t seem to have access to any workspaces.'],
+          [listView, () => this.wsList()],
+          () => this.wsGrid()
```
Comment (MattPutnam) [inline src/pages/workspaces/List.js:168]: Is it possible for `() => this.foo()` to be `this.foo`?

**Review by MattPutnam — APPROVED** [review]
Seems like a significant improvement to me

File: src/pages/workspaces/List.js (hunk 1 above)
Comment (zarsky-broad) [reply src/pages/workspaces/List.js:168]: No, I tried that first. It loses `this`

Comment (zarsky-broad) [reply src/pages/workspaces/List.js:168]: (those methods call `this` inside themselves)

File: src/libs/utils.js [hunk 2]
```diff
@@ -34,3 +37,28 @@ export const makePrettyDate = function(dateString) {
 }
 
 export const workspaceAccessLevels = ['NO ACCESS', 'READER', 'WRITER', 'OWNER', 'PROJECT_OWNER']
+
+export const log = function(arg) {
+  console.log(arg)
```
Comment (dmohs) [inline src/libs/utils.js:42]: Unlike this function in FireCloud, the reported line is always this line, regardless of the location of the call. For this reason, I'd suggest allowing multiple arguments so I can add a string to the front and hopefully have an easier time figuring out which log statement this is in an involved debugging session.

File: src/pages/workspaces/List.js (hunk 1 above)
Comment (dmohs) [reply src/pages/workspaces/List.js:168]: Not that I'm suggesting it, but couldn't you also do this?
```js
this.wsGrid.bind(this)
```

Comment (MattPutnam) [reply src/pages/workspaces/List.js:168]: Ah, too bad.

File: src/pages/workspaces/workspace/Dashboard.js [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
     const { workspace, failure, modal } = this.state
 
-    return _.isEmpty(workspace) ?
-      failure ?
-        `Couldn't load workspace details: ${failure}` :
-        spinner({ style: { marginTop: '1rem' } }) :
-      div({ style: { margin: '1rem' } }, [
+    return Utils.cond(
+      [failure, `Couldn't load workspace details: ${failure}`],
+      [!workspace, spinner({ style: { marginTop: '1rem' } })],
```
Comment (dmohs) [inline src/pages/workspaces/workspace/Dashboard.js:24]: I consider the `render` method in React to be similar to a tight loop in performance critical code. Despite the lack of elegance, I'd prefer the values to always be functions so only the truthy case is executed.

I do like the choice of using lists to differentiate the pairs.

File: src/pages/workspaces/List.js (hunk 1 above)
Comment (zarsky-broad) [reply src/pages/workspaces/List.js:168]: Yup, that'd work too @dmohs

File: src/pages/workspaces/workspace/Dashboard.js (hunk 3 above)
Comment (zarsky-broad) [reply src/pages/workspaces/workspace/Dashboard.js:24]: Ok. I'm going to make an exception when what's returned is a non-templated string.

File: src/libs/utils.js (hunk 2 above)
Comment (zarsky-broad) [reply src/libs/utils.js:42]: The issue with that is knowing which thing to return. I can include a stack trace.

Comment (zarsky-broad) [reply src/libs/utils.js:42]: Ah, you return the last thing. OK.

File: src/pages/workspaces/workspace/Dashboard.js (hunk 3 above)
Comment (dmohs) [reply src/pages/workspaces/workspace/Dashboard.js:24]: That does seem reasonable, but then you can't assert that the "value" is a function and help prevent programmer error.

**Pull request merged**